everything_else/reproducing.m

```diff
@@ -349,6 +349,14 @@
 %% Pg. 55, SCM-BCR
 %% Pg. 56, Decomposition into symmetric and anti-symmetric parts
 
+b = dec2bin(2^4-1:-1:0)-'0';
+b1 = b(:,1); b2 = b(:,2); b3 = b(:,3); b4 = b(:,4);
+
+f=b1.*b2 + b2.*b3 + b3.*b4 - 4*b1.*b2.*b3;
+
+f_sym = (1/2)*(f + f.');
+f_anti = (1/2)*(f - f.');
+
```

Comment:
This makes no sense. I don't see why you did `f.'`.

Reply:
This is to make the transpose of the matrix.

Reply:
@ndattani my alternative solution is below. I will add that to the code if you approve.

% f_sym = (1/2)*(f(b) + (1-f(b))
f_sym = (1/2)*(f + (1-f))

% f_anti = (1/2)*(f(b) - (1-f(b))
f_anti = (1/2)*(f - (1-f))

Reply:
What made you think that it would be a good idea to do the transpose?

Reply:
I looked into how decomposition into symmetric and antisymmetric would be and I see there are 2 ways to do it. One this by adding and subtracting transpose into f, divided by two. Source: https://www.mathworks.com/matlabcentral/answers/401295-how-to-find-the-symmetric-and-skew-symmetric-part-of-a-specified-matrix. 

The other is from the paper attached to the part in the book: https://www.maths.lth.se/matematiklth/vision/publdb/reports/pdf/kahl-strandmark-iccv-11.pdf. Notation: f - (1-transpose of f), divided by 2. 

Can you please explain the difference between those 2? Thank you.

Reply:
The second example doesn't use the transpose.

Reply:
It seems like I confused myself somewhere, my apologies. I will fix that tomorrow since I currently have to prepare to move.